embassy-stm32/src/rcc/g4.rs

```diff
@@ -73,39 +65,53 @@ pub enum Clock48MhzSrc {
 }
 
 /// Clocks configutation
+#[non_exhaustive]
 pub struct Config {
-    pub mux: ClockSrc,
+    pub hsi: bool,
+    pub hse: Option<Hse>,
+    pub sys: Sysclk,
+    pub hsi48: Option<super::Hsi48Config>,
+
+    pub pll: Option<Pll>,
+
+    /// Iff PLL is requested as the main clock source in the `mux` field then the PLL configuration
+    /// MUST turn on the PLLR output.
     pub ahb_pre: AHBPrescaler,
     pub apb1_pre: APBPrescaler,
     pub apb2_pre: APBPrescaler,
     pub low_power_run: bool,
-    /// Iff PLL is requested as the main clock source in the `mux` field then the PLL configuration
-    /// MUST turn on the PLLR output.
-    pub pll: Option<Pll>,
+
     /// Sets the clock source for the 48MHz clock used by the USB and RNG peripherals.
-    pub clock_48mhz_src: Option<Clock48MhzSrc>,
+    pub clk48_src: Option<Clock48MhzSrc>,
+
+    pub ls: super::LsConfig,
+
     pub adc12_clock_source: AdcClockSource,
     pub adc345_clock_source: AdcClockSource,
     pub fdcan_clock_source: FdCanClockSource,
 
-    pub ls: super::LsConfig,
+    pub boost: bool,
 }
 
 impl Default for Config {
     #[inline]
     fn default() -> Config {
         Config {
-            mux: ClockSrc::HSI,
+            hsi: true,
+            hse: None,
+            sys: Sysclk::HSI,
+            hsi48: Some(Default::default()),
+            pll: None,
             ahb_pre: AHBPrescaler::DIV1,
             apb1_pre: APBPrescaler::DIV1,
             apb2_pre: APBPrescaler::DIV1,
             low_power_run: false,
-            pll: None,
-            clock_48mhz_src: Some(Clock48MhzSrc::Hsi48(Default::default())),
+            clk48_src: Some(Clock48MhzSrc::Hsi48(Default::default())),
```

Comment:
the `Clock48MhzSrc::Hsi48` variant should not contain the HSI48 config, since it's already a separate field in the confg.